syft/source/directory_resolver.go

```diff
@@ -95,7 +92,7 @@ func (r *directoryResolver) indexTree(root string, stager *progress.Stage) ([]st
 	fi, err := os.Stat(root)
 	if err != nil && fi != nil && !fi.IsDir() {
 		// note: we want to index the path regardless of an error stat-ing the path
-		newRoot := r.indexPath(root, fi, nil)
+		newRoot, _ := r.indexPath(root, fi, nil)
```

Comment:
Do we still wan to totally drop the error here for the r.indexPath? Since we're surfacing the error maybe want something where it could be used in an `if err != nil { warn/debug }` rather than `fatal`

Reply:
The only "error" being returned is `fs.SkipDir`, which is a special thing to return during the walk that prevents traversal of subdirectories. This was ignoring anyone  errors already, effectively.